### PR TITLE
test: Add transfer-xudt integration tests

### PR DESCRIPTION
**Pull request by Dawn-githup**
### Description:
 - Write tests to make sure transfer-xudt works as expected. These tests will verify if transfers happen correctly.

### CODE REVIEW

**Review by duanyytop — APPROVED** [review]

File: tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    console.info(`xUDT asset has been minted or transferred and tx hash is ${txHash}`);
+    console.info(`explorer: https://pudge.explorer.nervos.org/transaction/${txHash}`);
+  });
+};
+
+const XUDT_TOKEN_INFO: RgbppTokenInfo = {
+  decimal: 8,
+  name: 'XUDT Test Token',
+  symbol: 'PDD',
```
Comment (Flouse) [inline tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:141]: The xUDT issued by the `fromLock` is [symbol: ccli](https://pudge.explorer.nervos.org/xudt/0x14f4972e17dd926f7ce55b3df5b1a31ba6e669a4ea7846e4f064535c1f50dac3) not `PDD`.


> [!NOTE]
> Only the first uniue cell combined with the xUDT in the same transaction is valid.
> a.k.a. The first transaction containing a unique cell will definitely declare the corresponding xUDT.
> See https://pudge.explorer.nervos.org/xudt/0x14f4972e17dd926f7ce55b3df5b1a31ba6e669a4ea7846e4f064535c1f50dac3

File: tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+};
+
+transferXudt({
+  // The xudtType comes from 1-issue-xudt
+  xudtType: {
+    codeHash: readStepLog('xUDT-type-script').codeHash,
+    hashType: readStepLog('xUDT-type-script').hashType,
+    args: readStepLog('xUDT-type-script').args,
+  },
+  receivers: [
```
Comment (Flouse) [inline tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:151]: ### Suggestion
test more than one receviers with different transferAmount

**Review by Flouse — CHANGES_REQUESTED** [review]

File: tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts (hunk 1 above)
Comment (Dawn-githup) [reply tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:141]: ### What I defined in the integration test is 'pdd'
https://github.com/ckb-cell/rgbpp-sdk/blob/2d634abae666610d53de03a9664d3e43fee1bef3/tests/rgbpp/xudt/xudt-on-ckb/1-issue-xudt.ts#L115C1-L117C3

Comment (duanyytop) [reply tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:141]: The same owner lock script is used to issue multiple xUDTs whose type scripts args are owner lock hash.

So the second issue transaction (PDD) will be regarded as re-issuing CCLI for CKB explorer.

Comment (Flouse) [reply tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:141]: > ### What I defined in the integration test is 'PDD'

See this tx: [0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9](https://pudge.explorer.nervos.org/transaction/0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9)

The xUDT was first defined as `ccli` on 2024/05/30 17:11:12

Comment (Dawn-githup) [reply tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:141]: > See this tx: [0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9](https://pudge.explorer.nervos.org/transaction/0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9)
> 
> The xUDT was first defined as `ccli` on 2024/05/30 17:11:12

Please check this transaction. 

https://pudge.explorer.nervos.org/transaction/0x6a35da16ab1198008545c78b91abe22999f0dc823055553a13d7de29f3063111
The example you provided is not the same lock.

Comment (Flouse) [reply tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts:141]: Ok, you are right.

**Review by Flouse — DISMISSED** [review]

**Review by duanyytop — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Pull request merged**